src/Compiler/test/Microsoft.NET.Sdk.Razor.SourceGenerators.Tests/RazorSourceGeneratorTests.TagHelpers.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.NET.Sdk.Razor.SourceGenerators;
+
+public partial class RazorSourceGeneratorTests
```

Comment:
I generally like the approach, though I do think this should be a separate test type. It's easier to work with.

Reply:
Easiest I can do this is add a nested class. If that doesn't sound good, I will refactor somehow to share the source generator testing utilities.

Reply:
Let's do the refactor. Single classes can be difficult from a number of tooling aspects.